proto/pocket/service/relay.proto

```diff
@@ -40,10 +41,22 @@ message JSONRPCRequestPayload {
     uint32 id = 1; // Identifier established by the Client to create context for the request.
     string jsonrpc = 2; // Version of JSON-RPC. Must be exactly "2.0".
     string method = 3; // Method being invoked on the server.
-    // TODO_RESEARCH: Find out why the params being a map causes errors
-    // should they be a list of maps?
-    //map<string, string> params = 4; // Parameters for the method. https://www.jsonrpc.org/specification#parameter_structures
-    repeated string params = 4; // Parameters for the method. https://www.jsonrpc.org/specification#parameter_structures
+    oneof params {
+        JSONRPCRequestPayloadParamsList list_params = 4 [(gogoproto.jsontag) = "params"];
+        JSONRPCRequestPayloadParamsMap map_params = 5 [(gogoproto.jsontag) = "params"];
+    }
+}
+
+// JSONRPCRequestPayloadParamsList contains the list of parameters for a JSON-RPC request.
+// TODO_RESEARCH: Will this always be a string list? Or should we use a more generic type?
```

Comment:
I thought about it and realized this could be a string of objects: `[obj1, obj2, etc..]`.

Wdyt of just making `prams` an `Any` like you suggested in discord?

Reply:
@Olshansk I think an approach could be storing the payload as its bytes. And just passing these directly. The downside of this is what if we wanted to do compute units?

Answer: we interim unmarshal without the params as compute units **should** probably only depend on the method being called (params shouldn't affect the method as the method's "weighting" would factor in the types of params it receives) and assign the weighting that way.

The reason I am against using `Any` in the protobuf is that the protobuf `Any` isn't 1-1 correspondance with the go `any` ie we cannot unmarhsal any type into an `Any` an any needs to be an `Any` type in go which the unmarhsaling step would again require some interim step of unmarshaling without the params and then a longer type checking.

Overall I think us trying to proto-ify the payload is probably the wrong path. What do we need it to be a protobuf for? Why cant be just directly pass the payload from the request to the service? Personally I think making the payload a protobuf is not for us to deal with - as we dont do anything with it, unlike the other field of the request structure.

WDYT?